test/extensions/clusters/redis/redis_cluster_test.cc

```diff
@@ -778,6 +856,46 @@ TEST_F(RedisClusterTest, AddressAsHostname) {
   EXPECT_EQ(0U, cluster_->info()->stats().update_failure_.value());
 }
 
+TEST_F(RedisClusterTest, AddressAsHostnameMultiplePrimaries) {
```

Comment:
@mattklein123 this is the unit test I've added. Do you have example of prior art on how to mock non-localhost machine in the response?

Reply:
Thank you for the pointer! I've updated the test to reflect checks for actual asynchronous resolution. I simplified the setup a bit to reduce test complexity. The test now initializes two slots with just primaries, but both their resolutions happen asynchronously. I verified that the test fails on `main` branch, and reproduces the SEGFAULT that we'd seen in our manual testing.